### ISSUE TITLE
Possible boundary condition in Array.slice

### ISSUE THREAD

**Issue by dterhorst** [elm-lang/core#892]
### Version
```elm
$ elm repl
---- elm-repl 0.18.0 -----------------------------------------------------------
 :help for help, :exit to exit, more at <https://github.com/elm-lang/elm-repl>
--------------------------------------------------------------------------------
```

### Code
```elm
> import Array
> Array.repeat 32 0 |> Array.slice 0 33
```

### Expected Result
```elm
Array.fromList [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
    : Array.Array number
```

### Actual Result
```elm
TypeError: Cannot read property 'height' of undefined
```

### Note

`Array.repeat 31 0 |> Array.slice 0 33` works.
`Array.repeat 32 0 |> Array.slice 0 33` fails.
`Array.repeat 33 0 |> Array.slice 0 33` works.

`Array.repeat 33 0 |> Array.slice 0 34` fails.

Given `Array.repeat length 0 |> Array.slice 0 limit`,
the problem seems to occur any time `32 <= length < limit`.

---

A similar problem occurs with `Array.fromList`:

```elm
> List.repeat 33 0 |> Array.fromList |> Array.slice 0 34
TypeError: Cannot read property 'height' of undefined
```

However, it only occurs when the length of the list
is `33 <= length < limit` as opposed to `32`, 
so there is something slightly different about that case.

**Comment by Skinney** [comment]
This should be fixed in Elm 0.19

**Comment by dterhorst** [comment]
Thanks. 👍
